code/modules/vending/mail.dm

```diff
@@ -0,0 +1,303 @@
+
+/obj/machinery/mailsorter/proc/update_state_after_sorting()
+	currentstate = STATE_IDLE
+	update_appearance()
+
+/obj/machinery/mailsorter/attackby(obj/item/thingy, mob/user, params)
+	if (istype(thingy, /obj/item/storage/bag/mail))
+		if (length(thingy.contents) < 1)
+			to_chat(user, span_warning("The [thingy] is empty!"))
+			return
+		var/loaded = 0
+		for (var/obj/item/mail in thingy.contents)
+			if (!(mail.item_flags & ABSTRACT) && \
+				!(mail.flags_1 & HOLOGRAM_1) && \
+				accept_check(mail) \
+			)
+				if (length(mail_list) + 1 > MAIL_CAPACITY )
+					to_chat(user, span_warning("There is no space for more mail in [src]!"))
+					return FALSE
+				else if (load(mail, user))
+					loaded++
+					mail_list += mail
+		if(loaded)
+			user.visible_message(span_notice("[user] loads \the [src] with \the [thingy]."), \
+			span_notice("You load \the [src] with \the [thingy]."))
+			if(length(thingy.contents))
+				to_chat(user, span_warning("Some items are refused."))
+			return TRUE
+		else
+			to_chat(user, span_warning("There is nothing in \the [thingy] to put in the [src]!"))
+			return FALSE
+	else if (istype(thingy, /obj/item/mail))
+		if (length(mail_list) + 1 > MAIL_CAPACITY )
+			to_chat(user, span_warning("There is no space for more mail in [src]!"))
+		else
+			thingy.forceMove(src)
+			mail_list += thingy
+			to_chat(user, span_notice("The [src] whizzles as it accepts the [thingy]."))
+	. = ..()
+
+/obj/machinery/mailsorter/proc/pick_mail(mob/user)
+	if(!length(mail_list))
+		return
+	var/obj/item/mail/mail_throw = tgui_input_list(user, "Choose the envelope to eject","Mail Sorting", mail_list)
+	if(!mail_throw)
+		return
+	currentstate = STATE_SORTING
+	update_appearance()
+	playsound(src, 'sound/machines/mail_sort.ogg', 20, TRUE)
+	addtimer(CALLBACK(src, PROC_REF(pick_envelope), user, mail_throw), 50)
+
+/obj/machinery/mailsorter/proc/pick_envelope(mob/user, obj/item/mail/mail_throw)
+	to_chat(user, span_notice("[src] reluctantly spits out [mail_throw]."))
+	var/turf/unload_turf = get_unload_turf()
+	mail_throw.forceMove(unload_turf)
+	mail_throw.throw_at(unload_turf, 2, 3)
+	mail_list -= mail_throw
+	currentstate = STATE_IDLE
+	update_appearance()
+
+/obj/machinery/mailsorter/proc/load(obj/item/thingy, mob/user)
+	if(ismob(thingy.loc))
+		var/mob/owner = thingy.loc
+		if(!owner.transferItemToLoc(thingy, src))
+			to_chat(owner, span_warning("\the [thingy] is stuck to your hand, you cannot put it in \the [src]!"))
+			return FALSE
+		return TRUE
+	else
+		if(thingy.loc.atom_storage)
+			return thingy.loc.atom_storage.attempt_remove(thingy, src, silent = TRUE)
+		else
+			thingy.forceMove(src)
+			return TRUE
+
+/obj/machinery/mailsorter/click_alt(mob/living/user)
+	if(!panel_open)
+		return CLICK_ACTION_BLOCKING
+	output_dir = turn(output_dir, -90)
+	to_chat(user, span_notice("You change [src]'s I/O settings, setting the output to [dir2text(output_dir)]."))
+	update_overlays()
```

Comment:
I didn't mean call this I meant use the UPDATE_OVERLAYS define as the arg to update_appearance